crates/uv-workspace/src/pyproject_mut.rs

```diff
@@ -82,6 +85,157 @@ impl ArrayEdit {
     }
 }
 
+/// The default version specifier when adding a dependency.
+// There is an implicit assumption that version numbers have three digits. While PEP 440
+// allows an arbitrary number of version digits, most projects stick to two or three digits, which
+// we can treat like a SemVer major.minor.patch.
+#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Eq, Serialize)]
+#[serde(rename_all = "kebab-case")]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub enum DependencyBoundDefault {
+    /// Only a lower bound, e.g., `>=1.2.3`.
+    #[default]
+    Lower,
+    /// Allow the same major version, similar to the semver caret, e.g., `>=1.2.3,<2.0.0`.
+    ///
+    /// Leading zeroes are skipped, e.g. `>=0.1.2,<0.2.0`.
+    Major,
+    /// Allow the same minor version, similar to the semver tilde does, `>=1.2.3,<1.3.0`.
```

Comment:
For consistency with the others
```suggestion
    /// Allow the same minor version, similar to the semver tilde, e.g., `>=1.2.3,<1.3.0`.
```